### ISSUE TITLE
Cannot upload to Artifactory repository with uppercase

### ISSUE THREAD

**Issue by fedterzi** [conan-io/conan-package-tools#167]
If I set `CONAN_UPLOAD=http://my_artifactory_server:8081/artifactory/api/conan/Artifactory_Repo` and try to upload a package, it will fail because the repository has been lowercased in the `ConanMultiPackager` constructor (file conan/packager.py lines 154-158):

```
                self.remotes = [remote.lower() for remote in self.remotes]
            else:
                self.remotes = self.remotes.lower()
        if self.upload:
            self.upload = self.upload.lower()
```
Error returned is:

```
ERROR: {
  "errors" : [ {
    "status" : 404,
    "message" : "Local repository 'artifactory_repo' not found"
  } ]
}

Invalid server response, check remote URL and try again. [Remote: upload_repo]
Traceback (most recent call last):
  File "build.py", line 28, in <module>
    builder.run()
  File "C:\Python27\lib\site-packages\conan\packager.py", line 427, in run
    self.upload_packages()
  File "C:\Python27\lib\site-packages\conan\packager.py", line 508, in upload_packages
    self.login("upload_repo")
  File "C:\Python27\lib\site-packages\conan\packager.py", line 500, in login
    raise Exception("Error with user credentials for remote %s" % remote_name)
Exception: Error with user credentials for remote upload_repo
```

Removing the 3 `.lower()` in those lines fixed the problem for me

**Comment by lasote** [comment]
Hi, thanks for reporting.
We have to review why are we converting to lowercase the remote URL, it feels very weird.

**Comment by lasote** [comment]
I released 0.14.0
Please, try again.

**Comment by fedterzi** [comment]
Thank you for your incredibly fast support! Works as a charm 🥇
